packages/edit-site/src/components/editor/use-adapt-editor-to-canvas.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect, useRegistry } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as editorStore } from '@wordpress/editor';
+import { useLayoutEffect } from '@wordpress/element';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+export function useAdaptEditorToCanvas( canvasMode ) {
```

Comment:
This was previously in the setCanvasMode action. But these are all "editor specific" changes, they should happen when the editor is rendered.